javascript/search/binary_search.js

```diff
@@ -0,0 +1,16 @@
+// Array is assumed to be sorted for the purposes of binary search. Return -1 if search item is not present
+function binarySearch(array, searchItem) {
+    return recursiveSearch(0, array.length - 1, array, searchItem);
+}
```

Comment:
With the latest JS syntax, you can actually write it like this:
```
const binarySearch(array, searchItem) => (
  recursiveSearch(0, array.length - 1, array, searchItem)
);
```
Not mandatory though, FYI

Reply:
Cool thanks edited and pushed